[bug] create_new_raster_from_base (utils/geoutils.py): bug if output array is singleband

create_new_raster_from_base fails in this situation:
```python
extract_archive(src="tests/data/spacenet.zip")
data = read_csv("tests/tiling/tiling_segmentation_binary-multiband_ci.csv")
ref_raster = Path(data[0]['tif'])
out_raster = ref_raster.parent / f"{ref_raster.stem}_copy.tif"
out_array = rasterio.open(ref_raster).read(1)  # read only one band
create_new_raster_from_base(input_raster=ref_raster, output_raster=out_raster, write_array=out_array)
```

Stack trace:
```
tests\utils\test_geoutils.py:37 (TestGeoutils.test_create_new_raster_from_base[bands_request0])
self = <test_geoutils.TestGeoutils object at 0x00000200FE481480>
bands_request = [1]

    def test_create_new_raster_from_base(self, bands_request) -> None:
        """Tests the 'create_new_raster_from_base' geo-utility"""
        extract_archive(src="tests/data/spacenet.zip")
        data = read_csv("tests/tiling/tiling_segmentation_binary-multiband_ci.csv")
        ref_raster = Path(data[0]['tif'])
        out_raster = ref_raster.parent / f"{ref_raster.stem}_copy.tif"
        out_array = rasterio.open(ref_raster).read(bands_request)
>       create_new_raster_from_base(input_raster=ref_raster, output_raster=out_raster, write_array=out_array)

tests\utils\test_geoutils.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
utils\geoutils.py:56: in create_new_raster_from_base
    dst.write(write_array[:, :], 1)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   ValueError: Source shape (1, 1, 426, 413) is inconsistent with given indexes 1

rasterio\_io.pyx:1695: ValueError
```

It seems the solution would simply be to remove the [if statement at end of function](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/geoutils.py#L55):
```python
        if count == 1:
            dst.write(write_array[:, :], 1)
        else:
            dst.write(write_array)
```
would become
```python
        dst.write(write_array)
```

--- a/tests/utils/test_geoutils.py
+++ b/tests/utils/test_geoutils.py
@@ -1,3 +1,4 @@
+import os
 from pathlib import Path
 from typing import List
 
@@ -43,3 +44,4 @@ class TestGeoutils(object):
         out_raster = ref_raster.parent / f"{ref_raster.stem}_copy.tif"
         out_array = rasterio.open(ref_raster).read(bands_request)
         create_new_raster_from_base(input_raster=ref_raster, output_raster=out_raster, write_array=out_array)
+        os.remove(out_raster)

--- a/tests/utils/test_geoutils.py
+++ b/tests/utils/test_geoutils.py
@@ -1,8 +1,14 @@
+from pathlib import Path
+from typing import List
+
 import numpy as np
+import pytest
 import rasterio
+from _pytest.fixtures import SubRequest
 from torchgeo.datasets.utils import extract_archive
 
 from dataset.aoi import AOI
+from utils.geoutils import create_new_raster_from_base
 from utils.utils import read_csv
 
 
@@ -22,3 +28,18 @@ class TestGeoutils(object):
         # make sure first band in multiband VRT is identical to source R band
         assert np.all(src_red_np == dest_red_np)
         aoi.close_raster()
+
+    @pytest.fixture(
+        params=[[1], [1, 2], [1, 2, 3]]
+    )
+    def bands_request(self, request: SubRequest) -> List:
+        return request.param
+
+    def test_create_new_raster_from_base_bands(self, bands_request) -> None:
+        """Tests the 'create_new_raster_from_base' geo-utility for different output bands number"""
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/tiling/tiling_segmentation_binary-multiband_ci.csv")
+        ref_raster = Path(data[0]['tif'])
+        out_raster = ref_raster.parent / f"{ref_raster.stem}_copy.tif"
+        out_array = rasterio.open(ref_raster).read(bands_request)
+        create_new_raster_from_base(input_raster=ref_raster, output_raster=out_raster, write_array=out_array)

--- a/utils/geoutils.py
+++ b/utils/geoutils.py
@@ -52,10 +52,7 @@ def create_new_raster_from_base(input_raster, output_raster, write_array):
                        crs=src.crs,
                        dtype=np.uint8,
                        transform=src.transform) as dst:
-        if count == 1:
-            dst.write(write_array[:, :], 1)
-        else:
-            dst.write(write_array)
+        dst.write(write_array)
 
 
 def get_key_recursive(key, config):

--- a/utils/geoutils.py
+++ b/utils/geoutils.py
@@ -51,7 +51,8 @@ def create_new_raster_from_base(input_raster, output_raster, write_array):
                        count=count,
                        crs=src.crs,
                        dtype=np.uint8,
-                       transform=src.transform) as dst:
+                       transform=src.transform,
+                       compress='lzw') as dst:
         dst.write(write_array)
 
 

